postgraas_server/management_resources.py

```diff
@@ -86,14 +86,27 @@ def delete(self, id):
                     host=current_app.postgraas_backend.master_hostname,
                     port=entity.port,
                     dbname=entity.db_name
-            ):
+            ) as conn:
                 pass
         except Exception as ex:
             return_code = 401 if 'authentication failed' in str(ex) else 500
             abort(return_code, status='failed',
                   msg='Could not connect to postgres instance: {}'.format(str(ex))
                   )
 
+        try:
+            cur = conn.cursor()
+            cur.execute("select count(pid) from pg_stat_activity where datname = %s ;", (entity.db_name,))
+            sessions = cur.fetchone()
+            if sessions[0] > 1:
+                raise HTTPError("Multiple sessions in database")
+        except Exception as ex:
```

Comment:
1. If you already _know_ that you're raising a specific `HTTPError`, you should not do a generic catch-all.
2. Why caputre the exception in `ex` if you do not re-use it?

Reply:
Replaced with a better solution